crowbar_framework/lib/crowbar/upgrade_status.rb

```diff
@@ -75,13 +75,19 @@ def current_step_state
       progress[:steps][current_step] || {}
     end
 
-    def start_step
+    # 'step' is name of the step user wants to start.
+    # If `out_of_order` is true, the step could be executed out of order
+    def start_step(step_name, out_of_order = false)
```

Comment:
aha :)

one more question:
wouldn't it be better to remove this `out_of_order` flag and embed the knowledge about steps which could be run out of order in the library itself?